src/Microsoft.AspNet.Mvc.ModelBinding/Binders/ArrayModelBinder.cs

```diff
@@ -2,16 +2,23 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
 using System.Threading.Tasks;
+using Microsoft.Framework.Internal;
 
 namespace Microsoft.AspNet.Mvc.ModelBinding
 {
+    /// <summary>
+    /// <see cref="IModelBinder"/> implementation for binding array values.
+    /// </summary>
+    /// <typeparam name="TElement">Type of elements in the array.</typeparam>
     public class ArrayModelBinder<TElement> : CollectionModelBinder<TElement>
     {
-        public override Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContext)
+        /// <inheritdoc />
+        public override Task<ModelBindingResult> BindModelAsync([NotNull] ModelBindingContext bindingContext)
         {
-            if (bindingContext.ModelMetadata.IsReadOnly)
+            if (bindingContext.Model == null && bindingContext.ModelMetadata.IsReadOnly)
```

Comment:
I kinda figure that you're right, but maybe it deserves a comment.. I read this code and it seems to me that this should be `||` instead of `&&`
- we don't want to try and update a null model
- we don't want to update a readonly model